### PR TITLE
[PLUGIN-954 & PLUGIN-1256] set cmek key in BQ job in BQExecute only when storing results in a table

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-954](https://cdap.atlassian.net/browse/PLUGIN-954) - Plugin Config Improvement suggested by @bdmogal.
[PLUGIN-1256](https://cdap.atlassian.net/browse/PLUGIN-1256) - Do not set cmek key by default in BQ job in BQ execute plugin because it causes job with ddl statements to fail.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 1]
```diff
@@ -279,6 +287,10 @@ public String getSql() {
       return sql;
     }
 
+    public Boolean getStoreResults() {
+      return storeResults != null && storeResults;
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java]: to be backward compatible , we should take null as true.

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 2]
```diff
@@ -229,9 +231,9 @@ public static final class Config extends AbstractBigQueryActionConfig {
     @Name(NAME_CMEK_KEY)
     @Macro
     @Nullable
-    @Description("The GCP customer managed encryption key (CMEK) name used to encrypt data written to " +
-      "any dataset or table created by the plugin. If the dataset or table already exists, this is ignored. More " +
-      "information can be found at https://cloud.google.com/data-fusion/docs/how-to/customer-managed-encryption-keys")
+    @Description("The GCP customer managed encryption key (CMEK) name used to encrypt data written to the " +
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:235]: "The GCP customer managed encryption key (CMEK) name used to encrypt the query results. It's only applicable when users choose to store the query results in a BigQuery table.  More information can be found at " +
      "https://cloud.google.com/data-fusion/docs/how-to/customer-managed-encryption-keys"

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 3]
```diff
@@ -115,14 +115,15 @@ public void run(ActionContext context) throws Exception {
...
     //create dataset to store the results if not exists
-    String datasetName = config.getDataset();
-    String tableName = config.getTable();
-    String datasetProjectId = config.getDatasetProject();
     if (!Strings.isNullOrEmpty(datasetName) && !Strings.isNullOrEmpty(tableName)) {
       BigQuerySinkUtils.createDatasetIfNotExists(bigQuery, DatasetId.of(datasetProjectId, datasetName),
                                                  config.getLocation(), cmekKeyName,
                                                  () -> String.format("Unable to create BigQuery dataset '%s.%s'",
                                                                      datasetProjectId, datasetName));
+      if (cmekKeyName != null) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:124]: should this cmekKey to be set before createDatasets ? or it has not impact on datasets creation ?

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:124]: yeah, it has no impact on dataset creation, that follows a different codepath.

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 4]
```diff
@@ -279,6 +289,10 @@ public String getSql() {
       return sql;
     }
 
+    public Boolean getStoreResults() {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:292]: I don't see `getStoreResults` is used at backend, is it purely used for UI ? Do you need to make sure only when `getStoreResults` is true, you store the result in the table ?

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:292]: addressed it in `review comments` commit https://github.com/data-integrations/google-cloud/pull/1042/files#diff-9ccb1a50b6e5a7a6b2c016c6aaa7605605d66e7e6bfb5e4e95a16ac7c35caa89R118

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java [hunk 5]
```diff
@@ -86,14 +87,13 @@ public void run(ActionContext context) throws Exception {
...
-    }
 
     // Save the results of the query to a permanent table.
-    if (config.getDataset() != null && config.getTable() != null) {
-      builder.setDestinationTable(TableId.of(config.getDataset(), config.getTable()));
+    String datasetName = config.getDataset();
+    String tableName = config.getTable();
+    String datasetProjectId = config.getDatasetProject();
+    if (datasetProjectId != null && datasetName != null && tableName != null) {
+      builder.setDestinationTable(TableId.of(datasetProjectId, datasetName, tableName));
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:96]: should this be guarded by getStoreResults too ?

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java:96]: ohh, yeah missed it

**Review by seanzhougoogle — APPROVED** [review]

**Pull request merged**